code/datums/action.dm

```diff
@@ -764,21 +823,29 @@
 /datum/action/cooldown/proc/PreActivate(atom/target)
 	if(SEND_SIGNAL(owner, COMSIG_ABILITY_STARTED, src) & COMPONENT_BLOCK_ABILITY_START)
 		return
+	StartCooldown(360 SECONDS, 360 SECONDS)
 	. = Activate(target)
+	StartCooldown()
 	SEND_SIGNAL(owner, COMSIG_ABILITY_FINISHED, src)
 
-/// To be implemented by subtypes
+/// To be implemented by subtypes (if not generic)
 /datum/action/cooldown/proc/Activate(atom/target)
-	return
+	var/total_delay = 0
+	for(var/datum/action/cooldown/ability as anything in initialized_actions)
+		if(ability.initialized_actions.len > 0)
+			ability.initialized_actions = list()
+		addtimer(CALLBACK(ability, .proc/Activate, target), total_delay)
+		total_delay += initialized_actions[ability]
 
 /datum/action/cooldown/UpdateButton(atom/movable/screen/movable/action_button/button, status_only = FALSE, force = FALSE)
 	. = ..()
 	if(!button)
 		return
 	var/time_left = max(next_use_time - world.time, 0)
-	if(text_cooldown)
+	if(text_cooldown && time_left < 180 SECONDS)
+		// don't display cooldown if its very long
 		button.maptext = MAPTEXT("<b>[round(time_left/10, 0.1)]</b>")
-	if(!owner || time_left == 0)
+	if(!owner || time_left == 0 || time_left > 180 SECONDS)
```

Comment:
This `180 SECONDS` number could probably be a define, given they're the same use (preventing it from displaying super long cooldown text)